addons/account/models/res_partner_bank.py

```diff
@@ -206,6 +206,17 @@ def write(self, vals):
                     initial_values['partner_id']._message_log(body=msg, tracking_value_ids=tracking_value_ids)
         return res
 
+    def default_get(self, fields_list):
+        if 'acc_number' not in fields_list:
+            return super().default_get(fields_list)
+
+        # When create & edit, `name` could be used to pass (in the context) the
+        # value input by the user. However, we want to set the default value of
+        # `acc_number` variable instead.
+        default_acc_number = self._context.get('default_acc_number', False) or self._context.get('default_name', False)
+        ctx_self = self.with_context(default_acc_number=default_acc_number)
+        return super(ResPartnerBank, ctx_self).default_get(fields_list)
```

Comment:
I would say that you could put that in a single line, as you don't reuse the ctx_self elsewhere